toolchain/base/mem_usage.h

```diff
@@ -47,27 +47,32 @@ class MemUsage {
                           .reserved_bytes = reserved_bytes});
   }
 
-  // Adds usage tracking for an allocator.
-  auto Add(std::string label, const llvm::BumpPtrAllocator& allocator) -> void {
+  // Adds memory usage for an allocator, typically `llvm::BumpPtrAllocator`.
+  template <class Allocator>
+    requires requires(const Allocator& allocator) {
```

Comment:
I see there's different kind of BumpPtrAllocators in llvm, I guess so far we've only used the default parameters. This requires clause would match them all, but it might be preferable to write this as:

```cc
  template <typename AllocatorT, size_t SlabSize, size_t SizeThreshold,
            size_t GrowthDelay>
  auto Collect(std::string label,
               const llvm::BumpPtrAllocatorImpl<
                   AllocatorT, SlabSize, SizeThreshold, GrowthDelay>& allocator)
      -> void {
```

OTOH this ties the function more to the llvm template type API, which may be more likely to break (though seems still unlikely) when rolling LLVM.

WDYT?

Reply:
Ok, done with `same_as`. PTAL